src/hook/useCardInfo.tsx

```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+import type { CardInfo } from '../types';
+
+const useCardInfo = () => {
+  const [cardInfo, setCardInfo] = useState<CardInfo[]>([]);
+
+  const registerNewCard = ({ cardNumber, expirationDate, cardOwnerName, selectedCard, cardNickName }: CardInfo) => {
+    setCardInfo([...cardInfo, { cardNumber, expirationDate, cardOwnerName, selectedCard, cardNickName }]);
+  };
```

Comment:
```suggestion
  const registerNewCard = useCallback((newCardInfo: CardInfo) => {
    setCardInfo((prevCardInfo) => [...prevCardInfo, newCardInfo]);
  }, []);
```

이렇게 사용하면 함수가 재사용 되면서 깔끔해지겠죠!

Reply:
오홋...! useCallback을 이용하면서 성능 향상도 하고 또 재사용도 가능하군여

감사합니당 ^^